File: rb/features.rb
# Isports SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IsportsFeatures
  def self.make_feature(name)
    case name
    when "base"
      IsportsBaseFeature.new
    when "test"
      IsportsTestFeature.new
    else
      IsportsBaseFeature.new
    end
  end
end

File: py/isports_sdk/features.py
# Isports SDK feature factory

from isports_sdk.feature.base_feature import IsportsBaseFeature
from isports_sdk.feature.test_feature import IsportsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IsportsBaseFeature(),
        "test": lambda: IsportsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
